easybuild/framework/easyblock.py

```diff
@@ -947,6 +947,10 @@ def make_module_dep(self, unload_info=None):
         self.log.debug("List of excluded deps: %s", excluded_deps)
 
         deps = [d for d in deps if d not in excluded_deps]
+        for dep in excluded_deps:
+            excluded_dep_deps = dependencies_for(dep, self.modules_tool)
```

Comment:
Ok, if it's already done elsewhere I'm fine with it (I just don't know where out of hand obviously).

As for speed this is a one-time operation so it would be ok I think.